tests/WebAppCallsWebApiCallsGraph/TodoListService/Models/TodoItem.cs

```diff
@@ -1,14 +1,19 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Text.Json.Serialization;
+
 namespace TodoListService.Models
 {
     public class Todo
     {
+        [JsonPropertyName("id")]
```

Comment:
@pmaytak
Why do we need to do that in the samples? Has anything changed?
Or is it to be cleaner?

Reply:
We don't have to. Yea, just so it's cleaner and more consistent.

Reply:
Also more robust, because without the attributes the serializer has to use the name of the properties, which a developer might change inadvertedly.

Reply:
I misunderstood the question. Let me rephrase. We don't have to replace Newtonsoft. I just did it so it's cleaner and more consistent with Identity Web. I chose to put attributes on properties because System.Text.Json does case-sensitive deserialization by default, so I just specified that JSON propeties in this object are lowercase. Otherwise I would have to pass deserialization options into every deserealize method.

Reply:
What if a customer doesn't do this? We need to document this and have it in the samples then.

---
In reply to: [436013645](https://github.com/AzureAD/microsoft-identity-web/pull/188#discussion_r436013645) [](ancestors = 436013645)

Reply:
Hmm. Good point. I could:
1. Revert back to using Newtonsoft - it's more flexible.
2. Instead of the `JsonPropertyName` I can just pass Json options to the deserealize method - it's a bit more straightforward.
3. If I keep it this way, where should I document this? It would probably be mostly just a link to the MS docs, since they explain it well.

Reply:
@pmaytak I'm not sure. @jmprieur what are your thoughts on this?

Reply:
How would option 2 look like, @pmaytak. It seems to me I prefer it better, but I've a hard time figuring out what it would mean?

Also, do we need to modify the Project templates?

Reply:
@jmprieur 
1. Something like:
```csharp
            var options = new JsonSerializerOptions
            {
                PropertyNameCaseInsensitive = true,
            };
            JsonSerializer.Deserialize<Todo>(content, options);
```
2. I didn't find any references to Newtonsoft in the projects.

Reply:
Indeed, @pmaytak, there are no references to Newtonsoft in the service (Web API) because the serialization is handled by ASP.NET core, hence my surprise (and me being worried) that the behavior would have changed and that we'd need to add these attribute to the samples POCO objects.

ASP.NET core handles the deserialization/serialization automatically in the controller based on the attributes, for instance `[FormBody]`

https://github.com/AzureAD/microsoft-identity-web/blob/68beba144a1b9b7c44e1ebaf34c048b1f69cf838/tests/B2CWebAppCallsWebApi/TodoListService/Controllers/TodoListController.cs#L59

I'm confused. Just to make sure I understand: do we really need to add the attributes or set the `JsonSerializerOptions`? or is it about demonstrating best practices, but not breaking anything.

Reply:
Yes, to your 3 points.

Interestingly, ASP.NET deserializaton from string body is case-**insensitive**. And ASP.NET serializes Todo to all lowercase property names before sending the result.

https://github.com/AzureAD/microsoft-identity-web/blob/5f70e638989f85d0d0b398eb185e77e301d7de2b/tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/TodoListController.cs#L63-L72

So yea, the JsonPropertyName attribute or JSON options are only when JsonSerializer class is used.